### PR TITLE
[Fix-14749][master] fix switch task  skip some task which should been executed

### LINKED ISSUE #14749: [Bug] [dolphinscheduler-master] In special cases, the switch task will skip task which need execute
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

In special cases, the switch task will skip task which need execute. As shown in the following figure, when switch result is s2,s1  have been executed,but the s3 task that should have been executed did not run regardless of the circumstances.
(当switch结果为s2任务,s1任务会被正常执行,但是本应任何情况下都应执行的s3任务没有被触发执行)
image

### What you expected to happen

s3 Task should be executed
(s3任务应该被触发执行)

### How to reproduce

Build the workflow shown above. when switch result is s2, s3 not been executed
image
image
image

### Anything else

_No response_

### Version

3.1.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LiuShuangBJ**
…  executed

## Purpose of the pull request

i try to fix  #14749 
我当时的场景是根据 上游节点的结果,判定 s1的数据是否完整,如果不完整,走s2步骤补充数据,然后执行s1及其以后的流程,但应用时发现,只要是走s2这条判断分支,s1会执行,但s1以后的任务节点不执行了,代码排查后发现,是如果是s2这条判断路径,在 
执行s3节点时会判断前置节点是否是跳过节点,此时s3会在如下代码片段中得到一个空set集合
image

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by caishunfeng** [comment]
Hi @LiuShuangBJ Please link a issue, describe the problem you fixed and how to reproduce? thanks.

**Comment by LiuShuangBJ** [comment]
已关联相关 issuse,在原issuse中有明确的复现步骤,以及代码级别的原因描述  @caishunfeng

**Comment by LiuShuangBJ** [comment]
已添加相应的issue,在issue中有明确的触发条件



At 2023-08-26 19:09:28, "caishunfeng" ***@***.***> wrote:

Hi @LiuShuangBJ Please link a issue, describe the problem you fixed and how to reproduce? thanks.

**Comment by qingwli** [comment]
Run 'mvn spotless:apply' to fix these violations.

**Comment by LiuShuangBJ** [comment]
代码格式化已完成,请审阅

**Comment by fuchanghai** [comment]
hi @LiuShuangBJ  it seem like similar to #14563  cc @caishunfeng

**Comment by fuchanghai** [comment]
hi @LiuShuangBJ  1.在s1 之前加一个 s0 ，再次运行不符合预期
2.s1如果依赖一个延迟任务，当从s2 分支执行到s1 时，延时任务还未执行，不包含在compeleteTask 中，导致s1 也会跳过

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java [hunk 1]
```diff
@@ -340,14 +340,18 @@ public static Set<Long> parsePostNodes(Long preNodeCode,
...
+                                              Map<Long, TaskNode> skipTaskNodeList,
+                                              Map<Long, TaskInstance> completeTaskList) {
         if (CollectionUtils.isEmpty(taskNode.getDepList())) {
             return false;
         }
         for (Long depNode : taskNode.getDepList()) {
             if (!skipTaskNodeList.containsKey(depNode)) {
                 return false;
             }
+            if (completeTaskList.containsKey(depNode)) {
```
Comment (fuchanghai) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java:352]: hi @LiuShuangBJ 1.在s1 之前加一个 s0 ，再次运行不符合预期
2.s1如果依赖一个延迟任务，当从s2 分支执行到s1 时，延时任务还未执行，不包含在compeleteTask 中，导致s1 也会跳过

**Review by fuchanghai — CHANGES_REQUESTED** [review]

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java (hunk 1 above)
Comment (LiuShuangBJ) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java:352]: yes ,and this problem had been closed by #14537, i will close this pr

**Pull request closed without merging**